### DISCUSSION TITLE
Address & Map Issues when entering meetings into Meeting Guide

### DISCUSSION THREAD

**Discussion by D23A74** [code4recovery/12-step-meeting-list#626] [Q&A] [answered]
Hi - I have never worked on webpages before so am very green at this.  Our District webpage is AADistrict23WI.org.  We are using GoDaddy as our host and I'm using Managed Workpress for the website.  I have two main issues right now that I'm requesting your help with:
1)  I am using Google Maps and when I enter the meeting location’s address into Wordpress, the proper map pops up for a split second and then disappears and is replaced by an error message that reads “Oops! Something went wrong!  This page didn't load Google Maps correctly. See the JavaScript console for technical details.”   I have no idea where or what the JavaScript console is.  And then, even stranger, the map feature usually works fine for users using the Meeting Guide app.  Why is this showing up on my Wordpress version?
2)  My second issue may or may not be connected:  The addresses for some meetings that I enter will not “take” or allow me to publish them if the system ‘thinks’ they’re wrong.  For instance, if there’s a Stout Street in both Rice Lake and Chetek, it keeps reverting my Chetek entry to Rice Lake - even though it has a different zip code and Region.  Also, if there is more than one Bethany Lutheran Church (for instance) but in two different towns, the systems reverts my entry back to the first one I entered.  Have you heard of this happening?  I have several entries to the Meeting Guide that I can’t publish correctly because of this.  I’ve also had problems with the system changing the “Region” to what it thinks it should be – even though I’ve manually entered it as some other region. 
     Any help that anyone can provide would be GREATLY appreciated!!  Also, please let me know if you need any additional information that I haven't yet provided.  Thank you!!
P.S.  I would provide some screen shots but the copy/paste function doesn't seem to work in this box.
https://aadistrict23wi.org/meetings/sunday-big-book/?tsml-day=any

**Comment by brianw-area24** [comment, marked as answer]
With your first issue, what the "Javascript Console" would tell you is
> Google Maps JavaScript API error: ApiNotActivatedMapError
> https://developers.google.com/maps/documentation/javascript/error-messages#api-not-activated-map-error

What this is saying is that your Google API key doesn't have the correct settings to be able to display the map. You have a couple of options to address this
1. Log into the Google console, and activate the "Maps JavaScript API". 
2. Login to https://www.mapbox.com and create an access token, and put this token into the "Import & Settings" page of the TSML plugin. This will allow your site to use Mapbox maps.
Personally, I find the interface of the Google console confusing, too many options. Therefore, unless you need to use Google maps, I'd probably recommend Mapbox for your site.

Next, also on the Import & Settings page, I would suggest you select "BETA - API Gateway - BETA" for the Address Geocoding Method. This may help with your second issue.

Here is an image of the bottom right-hand corner of the Import & Settings page of my test site.

![Screen Shot 2022-01-08 at 1 12 10 PM](https://user-images.githubusercontent.com/55821195/148656754-b81b0006-5738-4e56-a943-53afa65021ac.png)

**Reply by D23A74** [reply]
Thank you SO much, Brian!!  I did use Mapbox and selected BETA-API Gateway-BETA on the Imports & Settings page and the map problem is resolved!!  You are the best!!  This did not solve my 2nd issue but by experimenting, I think it is a system glitch.  The address I was trying was 216 Stout St, Chetek, WI 54728 (Region: Chetek).  It kept reverting to Rice Lake (vs Chetek).  So I tried enter 217 Stout St. and it accepted that for Chetek.  So then I tried "0216 Stout St" vs "216 Stout St" and it worked!!  It also dropped the zero so everything appears correctly now!!  Again, thank you so much for your help!  This was driving me absolutely mad!!  I am VERY grateful!
